File: 485_Max_Consecutive_Ones.java
/*
https://leetcode.com/problems/max-consecutive-ones/
Given a binary array, find the maximum number of consecutive 1s in this array.

Iterate over array, incrementing current consecutive count for every 1, resetting and updating max for every 0.
Time - O(n)
Space - O(1)
*/

public class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {

        int consec = 0;
        int maxConsec = 0;

        for (int num : nums) {
            if (num == 0) {
                maxConsec = Math.max(maxConsec, consec);
                consec = 0;
            } else {
                ++consec;
            }
        }
        return Math.max(maxConsec, consec);     // case of nums ending in 1s
    }
}

File: python_1_to_1000/485_Max_Consecutive_Ones.py
_author_ = 'jake'
_project_ = 'leetcode'

# https://leetcode.com/problems/max-consecutive-ones/
# Given a binary array, find the maximum number of consecutive 1s in this array.

# Iterate over nums. If we see a zero, update max_consecutive and reset consecutive. If we see a one, increment
# consecutive.
# Time - O(n)
# Space - O(1)

class Solution(object):
    def findMaxConsecutiveOnes(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        consecutive, max_consecutive = 0, 0

        for num in nums:
            if num == 0:
                max_consecutive = max(max_consecutive, consecutive)
                consecutive = 0
            else:
                consecutive += 1

        return max(max_consecutive, consecutive)    # check final sequence
